app/src/main/java/com/pavesid/androidacademy/ui/MainActivity.kt

```diff
@@ -36,9 +94,44 @@ class MainActivity : AppCompatActivity(), MoviesFragment.Listener {
         return super.onOptionsItemSelected(item)
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
-        _binding = null
+    override fun onAccuracyChanged(sensor: Sensor, accuracy: Int) {}
+
+    override fun onSensorChanged(event: SensorEvent) {
+
+        // The sensorEvent object is reused across calls to onSensorChanged().
+        // clone() gets a copy so the data doesn't change out from under us
+        when (event.sensor.type) {
+            TYPE_ACCELEROMETER ->
+                gravity = event.values.clone()
+            TYPE_MAGNETIC_FIELD ->
+                geomagnetic = event.values.clone()
+        }
+
+        val normalizerGravity =
+            sqrt(gravity[0] * gravity[0] + gravity[1] * gravity[1] + gravity[2] * gravity[2])
+
+        gravity[0] = gravity[0] / normalizerGravity
+        gravity[1] = gravity[1] / normalizerGravity
+        gravity[2] = gravity[2] / normalizerGravity
+
+        var rotation = when (thisDisplay?.rotation) {
+            Surface.ROTATION_0 ->
+                round(Math.toDegrees(atan2(gravity[0].toDouble(), gravity[1].toDouble())))
+            Surface.ROTATION_90 ->
+                round(Math.toDegrees(atan2(gravity[0].toDouble(), gravity[1].toDouble()))) - 90
+            Surface.ROTATION_180 ->
+                round(Math.toDegrees(atan2(gravity[0].toDouble(), gravity[1].toDouble())))
+            Surface.ROTATION_270 ->
+                round(Math.toDegrees(atan2(gravity[0].toDouble(), gravity[1].toDouble()))) + 90
+            else -> round(Math.toDegrees(atan2(gravity[0].toDouble(), gravity[1].toDouble())))
```

Comment:
maybe extract **round(Math.toDegrees(atan2(gravity[0].toDouble(), gravity[1].toDouble())))** to one place ?